guides/testing.md

```diff
@@ -20,60 +44,92 @@ Concrete test cases are defined in `it` blocks. An optional (but strongly recomm
-| `be_true`      | passes if actual is true (`== true`) |
-| `be_false`     | passes if actual is false (`== false`) |
-| `be_truthy`    | passes if actual is truthy (neither `nil` nor `false`) |
-| `be_falsey`    | passes if actual is falsy (`nil` or `false`) |
-| `be < value`  | passes if actual is lesser than *value* (`<`) |
-| `be <= value` | passes if actual is lesser than or equal *value* (`<=`) |
-| `be > value`  | passes if actual is greater than *value* (`>`) |
-| `be >= value` | passes if actual is greater than or equal *value* (`>=`) |
-| `be_close(expected, delta)` | passes if actual is within *delta* of *expected* |
-| `contain`      | passes if actual includes *expected* (`.includes?`) |
-| `match(value)` | passes if actual matches *value* (`=~`) |
-| `be_a(type)`   | passes if actual is of type *type* (`is_a?`) |
-| `expect_raises(&block)` | runs the block and passes if it raises an exception |
-| `expect_raises(klass, &block)` | runs the block and passes if it raises an exception of type *klass* |
-| `expect_raises(klass, message, &block)` | runs the block and passes if it raises an exception of type *klass* and the error message contains *message* |
+### Equivalence, Identity and Type
+```crystal
+actual.should eq(expected)    # passes if actual == expected
+actual.should be(expected)    # passes if actual.same?(expected)
+actual.should be_a(expected)  # passes if actual.is_a?(expected)
+actual.should be_nil          # passes if actual.nil?
+```
+
+### Truthiness
+```crystal
+actual.should be_true         # passes if actual == true
+actual.should be_false        # passes if actual == false
+actual.should be_truthy       # passes if actual is truthy (neither nil nor false)
+actual.should be_falsey       # passes if actual is falsey (nil or false)
+```
+
+### Comparisons
+```crystal
+actual.should be <  expected  # passes if actual <  expected
+actual.should be <= expected  # passes if actual <= expected
+actual.should be >  expected  # passes if actual >  expected
+actual.should be >= expected  # passes if actual >= expected
+```
+
+### Other matchers
+```crystal
+actual.should be_close(expected, delta) # passes if actual is within delta of expected:
+                                        # (actual - expected).abs <= delta
+actual.should contain(expected)         # passes if actual.includes?(expected)
+actual.should match(expected)           # passes if actual ==~ expected
+```
+
+### Expecting errors
+
+These matchers run a block and pass if it raises a certain exception.
+
+```crystal
+expect_raises do
+  # passes if this block raises any exception
+  # NOTE: This should be avoided because it can hide errors.
+end
+
+expect_raises(MyError) do
+  # passes if this block raises an exception of type MyError
+end
+
+expect_raises(MyError, "error message") do
+  # passes if this block raises an exception of type MyError
+  # and the error message contains "error message"
+end
+```
```

Comment:
There is also `expect_raises(MyError, /message/)` (with a regex instead of a string)